src/nccl_ofi_net.c

```diff
@@ -1093,8 +1104,80 @@ static ncclResult_t ofi_accept(void *listenComm, void **recvComm)
 	return ret;
 }
 
+static ncclResult_t ofi_regMr(void *comm, void *data, int size, int type,
+			      void **mhandle)
+{
+	struct fid_mr *mr_handle = NULL;
+	ncclResult_t ret = ncclSuccess;
+
+	/*
+	 * Let's assume the given Comm structure is for sendComm.
+	 * It doesn't matter as we will extract only dev from it
+	 */
+	sendComm_t *sComm = (sendComm_t *)comm;
+
+	/* Validate Comm */
+	if (OFI_UNLIKELY(sComm == NULL)) {
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("Invalid Comm object provided");
+		goto exit;
+	}
+
+	/* Validate type to be NCCL_PTR_HOST */
+	if (OFI_UNLIKELY(type != NCCL_PTR_HOST)) {
+		ret = ncclSystemError;
+		NCCL_OFI_WARN("Plugin supports registration of host buffers only");
+		goto exit;
+	}
+
+	/* Check if we do registration of local buffers */
+	if (!local_mr) {
+		goto exit;
+	}
+
+	ret = fi_mr_reg(nccl_ofi_component[sComm->dev]->domain, data, size,
+			FI_SEND | FI_RECV, 0, 0, 0, &mr_handle, NULL);
+	if (OFI_UNLIKELY(ret != 0)) {
+		NCCL_OFI_WARN("Unable to register memory for device %d. RC: %d, Error: %s",
+			      sComm->dev, fi_strerror(-ret));
+		ret = ncclSystemError;
+	}
+
+exit:
+	*mhandle = mr_handle;
```

Comment:
This should be `*mhandle = fi_mr_desc(mr_handle);` as the fi_tsend/fi_trecv functions require an fi_mr_desc pointer in their arguments.

Reply:
That is correct. Thanks for pointing out! I will make the change.